_posts/blog/2016-09-13-code-reviews.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: post
+title: "Handbook of manners: how to behave in a code review"
+description: Some thoughts on code reviews and the way they usually are and they should be performed.
+modified:
+categories: blog
+author: henning_gross
+excerpt: Code reviews are a common practise for knowledge distribution, learning and increasing quality - but do we perform them i a way that actually creates value?
+tags: [Mindset, Culture, Agile]
+date: 2016-09-13T17:49:00-04:00
+---
+
+In most companies code reviews are part of the process today. In theory they help distributing knowledge, onboarding new people, increase learning and quality.
+
+A code review, no matter if it is performed in a pairing manner or asynchronously is a situation in which one person inspects the work of another person.
+
+This can lead to the assumption that the reviewer has to accept the changes the reviewee wants to introduce and put the reviewer in a position of power over the reviewee who has to seek for approval.
+
+Agile development teams strive for a conception and identity in which everyone is equally important, equally worthy and hierarchies usually make no difference on functional matters (as long as the manager does not have to protect the business but thats another chapter).
+
+So how does a code-review ideally look like?
+
+Lets base on the assumption that every reviewer has the same intent - which is to convince the reviewee to follow the suggested changes (to improve the quality of the added code)?
+
+When this is the main intent, different approaches to code reviews can only be different strategies to achieve the same goal.
+
+The following strategies/behavioural advices can help to achieve this goal and at the same time improve the team spirit and culture.
+
+# Consult. Dont preach.
+
+The basic mindset in which an engineer should approach a code review for a change introduced by another engineer is the mindset of a consultant. Whatever you say or write: do it with caution. Sentences that start with "I would approach the problem this way..." or "I believe this way would be more efficient" are a good approach if you want to encourage the reviewee to accept your recommendations. Way better than "this solution is inappropriate" or "do it like this". Command-style comments are guaranteed to result in damage. Conflicts, drop of creativity and motivation, bad team spirit. Dont do it.
```

Comment:
Since sentences like "this solution is inappropriate" are [direct quotes] (https://www.grammarly.com/handbook/mechanics/capitalization/11/capitalization-the-first-word-in-a-sentence-that-i/) I would start them with a capital letter and put an exclamation mark at the end. This also makes them sound more powerful, and demanding, as I think you intended.